aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -972,7 +973,9 @@ static void apply_svd(Tensor& self, Tensor& U, Tensor& S, Tensor& VT,
     } else {
       lrwork = std::max(7 * mn * mn + 7 * mn, 2 * mx * mn + 2 *mn * mn + mn);
     }
-    rwork = at::empty({std::max(int64_t(1), lrwork)}, at::kInt);
+    // rwork is an array of floats or doubles depending on the type
+    auto rwork_multiplier = self_stype == at::kComplexDouble ? 2 : 1;
+    rwork = at::empty({rwork_multiplier * std::max(int64_t(1), lrwork)}, at::kInt);
```

Comment:
but you still set the dtype of rwork to int. shouldn't you set that to `c10::toValueType(self_stype)` (which gives the corresponding floating point type for complex)?

Reply:
Code is a bit ugly right now, as lapackSvd expects raw `int*` as rwork instead of a structured tensor. Let me work on the logic to make it look a bit prettier.